src/discussions/comments/data/slices.js

```diff
@@ -3,14 +3,28 @@ import { createSlice } from '@reduxjs/toolkit';
 import { RequestStatus } from '../../../data/constants';
 
 function normaliseComments(state, rawCommentData) {
-  const { threadCommentMap: threads, comments } = state;
+  const {
+    threadCommentMap: threads,
+    commentResponsesMap: responses,
+    comments,
+  } = state;
   rawCommentData.forEach(
     comment => {
-      if (!threads[comment.thread_id]) {
-        threads[comment.thread_id] = [];
-      }
-      if (!threads[comment.thread_id].includes(comment.id)) {
-        threads[comment.thread_id].push(comment.id);
+      if (comment.parent_id) {
+        if (!responses[comment.parent_id]) {
+          responses[comment.parent_id] = [];
+        }
+        if (!responses[comment.parent_id].includes(comment.id)) {
+          responses[comment.parent_id].push(comment.id);
+        }
```

Comment:
Using sets makes this simpler IMO:
```js
if (!responses[comment.parent_id]) {
  responses[comment.parent_id] = new Set();
}
responses[comment.parent_id].add(comment.id);
```

Reply:
I think I'm going to keep things the way they are. Making this a set means that we now can use `.map` on it like a regular array, so it will need to be converted to an array, or we'll need to use a different approach in the component. I'd rather that the components not need to be aware of the specific data structures inside the state.